File: trunk/PolarMath/src/org/macondo/polar/evaluation/time/PNN50.java
package org.macondo.polar.evaluation.time;

import org.macondo.polar.data.Training;
import org.macondo.polar.evaluation.Evaluation;

import java.util.List;

/**
 * <p></p>
 *
 * Date: 19.05.2008
 *
 * @author <a href="mailto:ktsibriy@gmail.com">Kirill Y. Tsibriy</a>
 */
public class PNN50 implements Evaluation<Integer> {
    public Integer evaluate(Training training) {
        final List<Integer> intervals = training.getIntervals();
        int pnn = 0;


        for (int i = 1; i < intervals.size(); i++) {
            Integer now = intervals.get(i);
            Integer before = intervals.get(i-1);

            if (Math.abs(now - before) >= 50) {
                pnn++;
            }
        }

        return (int)(((double) pnn) / (intervals.size() - 1) * 100);
    }
}

File: trunk/PolarMathCS/PolarMath/Evaluation/Time/PNN50.cs
using PolarMath.Data;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PolarMath.Evaluation.Time
{
    public class PNN50 : Evaluation<int>
    {
        public int evaluate(Training training) 
        {
            List<int> intervals = training.getIntervals();
            int pnn = 0;


            for (int i = 1; i < intervals.Count; i++) {
                int now = intervals.ElementAt(i);
                int before = intervals.ElementAt(i-1);

                if (Math.Abs(now - before) >= 50) {
                    pnn++;
                }
            }

            return (int)(((double) pnn) / (intervals.Count - 1) * 100);
        }
    }
}
